File: samples/Marketplace/Marketplace.Domain/Models/Trading/OrderItem.cs
// Copyright 2017 (c) [Denis Da Silva]. All rights reserved.
// See License.txt in the project root for license information.

using System;
using Prolix.Domain;

namespace Marketplace.Domain.Models.Trading
{
    public class OrderItem : ActiveModel
    {
        public decimal Quantity { get; set; }
        public decimal Amount { get; set; }

        public int? OrderId { get; set; }
        public int? ProductId { get; set; }

        public virtual Order Order { get; set; }
        public virtual Product Product { get; set; }
    }
}

File: samples/Marketplace/Marketplace.Web/app/models/OrderItem.ts
// Copyright 2017 (c) [Denis Da Silva]. All rights reserved.
// See License.txt in the project root for license information.

/// <reference path="../_references.ts" />

module App.Models {
    'use strict';

    export class OrderItem extends Model {
        public Quantity: number;
        public Price: number;
        public Amount: number;

        public ProductId: number;
        public ProductName: string;
    }
} 
